Bugzilla/User.pm

```diff
@@ -478,10 +482,25 @@ sub set_login {
     delete $self->{nick};
 }
 
+sub _generate_nickname {
+    my ($name, $login) = @_;
+    my ($nick) = extract_nicks($name);
+    if (!$nick) {
+        $nick = "";
+    }
+    return $nick;
+}
+
 sub set_name {
     my ($self, $name) = @_;
     $self->set('realname', $name);
     delete $self->{identity};
+    $self->set('nickname', _generate_nickname($name, $self->login));
+}
+
```

Comment:
'set_nick' because nick() is the accessor (with some historical-enforced caveats).
Future work might be making the column names and method names *sane* but that it is outside the scope of this enhancement.